modules/dxp/apps/commerce/headless/headless-commerce-bom-impl/src/main/java/com/liferay/headless/commerce/bom/internal/dto/v1_0/converter/DiagramDataDTOConverter.java

```diff
@@ -0,0 +1,175 @@
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.headless.commerce.bom.internal.dto.v1_0.converter;
+
+import com.liferay.commerce.product.model.CPAttachmentFileEntry;
+import com.liferay.commerce.product.model.CPInstance;
+import com.liferay.commerce.product.model.CProduct;
+import com.liferay.commerce.product.service.CPInstanceLocalService;
+import com.liferay.commerce.product.service.CPInstanceService;
+import com.liferay.commerce.product.service.CProductLocalService;
+import com.liferay.commerce.shop.by.diagram.model.CPDefinitionDiagramEntry;
+import com.liferay.commerce.shop.by.diagram.model.CPDefinitionDiagramSettings;
+import com.liferay.commerce.shop.by.diagram.service.CPDefinitionDiagramEntryService;
+import com.liferay.commerce.shop.by.diagram.service.CPDefinitionDiagramSettingsService;
+import com.liferay.document.library.kernel.util.DLUtil;
+import com.liferay.headless.commerce.bom.dto.v1_0.DiagramData;
+import com.liferay.headless.commerce.bom.dto.v1_0.Product;
+import com.liferay.headless.commerce.bom.dto.v1_0.Spot;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterContext;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(
+	enabled = false,
+	property = "model.class.name=cpDefinitionDiagramSettingsData",
+	service = {DiagramDataDTOConverter.class, DTOConverter.class}
+)
+public class DiagramDataDTOConverter
+	implements DTOConverter<CPDefinitionDiagramSettings, DiagramData> {
+
+	@Override
+	public String getContentType() {
+		return DiagramData.class.getSimpleName();
+	}
+
+	@Override
+	public DiagramData toDTO(DTOConverterContext dtoConverterContext)
+		throws Exception {
+
+		String image = StringPool.BLANK;
+
+		CPDefinitionDiagramSettings cpDefinitionDiagramSettings =
+			_cpDefinitionDiagramSettingsService.getCPDefinitionDiagramSettings(
+				(Long)dtoConverterContext.getId());
+
+		CPAttachmentFileEntry cpAttachmentFileEntry =
+			cpDefinitionDiagramSettings.fetchCPAttachmentFileEntry();
+
+		if (cpAttachmentFileEntry != null) {
+			FileEntry fileEntry = cpAttachmentFileEntry.getFileEntry();
+
+			image = DLUtil.getDownloadURL(
+				fileEntry, fileEntry.getFileVersion(), null, null);
+		}
+
+		String imageURL = image;
```

Comment:
Just call it imageURL
```suggestion
		String imageURL = StringPool.BLANK;

		CPDefinitionDiagramSettings cpDefinitionDiagramSettings =
			_cpDefinitionDiagramSettingsService.getCPDefinitionDiagramSettings(
				(Long)dtoConverterContext.getId());

		CPAttachmentFileEntry cpAttachmentFileEntry =
			cpDefinitionDiagramSettings.fetchCPAttachmentFileEntry();

		if (cpAttachmentFileEntry != null) {
			FileEntry fileEntry = cpAttachmentFileEntry.getFileEntry();

			imageURL = DLUtil.getDownloadURL(
				fileEntry, fileEntry.getFileVersion(), null, null);
		}
```